tests/PhpUnit/GuzzleAssertsTraitTest.php

```diff
@@ -206,6 +207,31 @@ public function testAssertRequestHeaderDoesNotMatch()
         }
     }
 
+    public function testAssertRequestQueryDoesNotMatch()
+    {
+        $query = [
+            'tags' => ['foo', '1'],
+        ];
+
+        $request = $this->createMockRequest('GET', '/api/pets', $this->getValidHeaders(), $this->getValidRequestBody(), $query);
+
+        try {
+            self::assertRequestMatch($request, $this->schemaManager);
+            self::fail('Expected ExpectationFailedException to be thrown');
+        } catch (ExpectationFailedException $e) {
+            self::assertEquals(
+                <<<EOF
+Failed asserting that {"tags":["foo",1]} is a valid request query.
+[limit] The property limit is required
+[tags[1]] Integer value found, but a string is required
```

Comment:
indeed, that's the same issue as form fields and its definitely not easy. I'll revert these changes and keep them apart.